File: python/src/shapes/base/models/rectangle.py
from typing import final

from shapes.base.commons import Dimensional


@final
class Rectangle(Dimensional):
    def __init__(self, width: float, height: float):
        dim_one: float = width if width > 0 else 1
        dim_two: float = height if height > 0 else 1
        self.width = dim_one
        self.height = dim_two

    def area(self) -> float:
        return self.width * self.height

    def perimeter(self) -> float:
        return 2 * (self.width + self.height)

File: ruby/lib/shapes/base/models/rectangle.rb
# frozen_string_literal: true

require "shapes/base/commons"

class Rectangle
    include Dimensional

    attr_accessor :width, :height

    def initialize(width, height)
        dim_one = width >= 0 ? width : 0
        dim_two = height >= 0 ? height : 0
        @width = dim_one
        @height = dim_two
    end

    def area
        @width * @height
    end

    def perimeter
        2 * (@width + @height)
    end
end
